lib/src/components/Swiper/index.tsx

```diff
@@ -0,0 +1,334 @@
+      key,
+    })
+  })
+
+  const firstPageToShow = centeredSlides
+    ? // if centeredSlides is true, we calculate which number is the middle page
+      Math.floor(numberOfPage / 2)
+    : // if centeredSlides is false, we calculate on which page the number in firstSlideToShow props is
+      Math.ceil(firstSlideToShow / currentSlidesPerView) - 1
+
+  const getArrowStates = useCallback(() => {
+    const sliderContainer = ref?.current
+    if (sliderContainer && !shouldLoop) {
+      const { offsetWidth, scrollLeft, scrollWidth } = sliderContainer
+      const isFirstPage = !(scrollLeft > spaceBetween)
+      const isLastPage = !(scrollWidth - (scrollLeft + offsetWidth) > spaceBetween)
+
+      setShowLeftArrow(!isFirstPage)
+      setShowRightArrow(!isLastPage)
+    } else {
+      setShowLeftArrow(true)
+      setShowRightArrow(true)
+    }
+  }, [ref, setShowLeftArrow, setShowRightArrow, shouldLoop, spaceBetween])
+
+  const updatePage = () => {
+    const sliderContainer = ref?.current
+    if (sliderContainer) {
+      const { children, offsetWidth, scrollLeft, scrollWidth } = sliderContainer
+      const childWidth = children?.[0]?.getBoundingClientRect()?.width
+
+      const isLastPage = !(scrollWidth - (scrollLeft + offsetWidth) > spaceBetween)
+
+      const nextPage = isLastPage
+        ? bullets.length - 1
+        : Math.round(scrollLeft / ((childWidth + spaceBetween) * currentSlidesPerView))
+
+      if (nextPage !== currentPage) {
+        setCurrentPage(nextPage)
+      }
+    }
+  }
+
+  const handleScroll = debounce(() => {
+    getArrowStates()
+    updatePage()
+  }, 100)
+
+  const goTo = useCallback(
+    (page: number, isFirstInit = false) => {
+      const sliderContainer = ref?.current
+      const childWidth = sliderContainer?.children?.[0]?.getBoundingClientRect()?.width
+
+      sliderContainer?.scrollTo({
+        // We don't want to have a scroll effect when we first render the swiper
+        behavior: !isFirstInit ? 'smooth' : 'auto',
+        left: page * (childWidth + spaceBetween) * currentSlidesPerView,
+        top: 0,
+      })
+    },
+    [currentSlidesPerView, spaceBetween, ref]
+  )
+
+  const goNext = useCallback(() => {
+    if (shouldLoop && isLastPage) {
+      goTo(0)
+    } else {
+      goTo(currentPage + 1)
+    }
+  }, [currentPage, goTo, isLastPage, shouldLoop])
+
+  const goPrev = useCallback(() => {
+    if (isFirstPage && shouldLoop) {
+      goTo(bullets.length - 1)
+    } else {
+      goTo(currentPage - 1)
+    }
+  }, [bullets.length, currentPage, goTo, isFirstPage, shouldLoop])
+
+  // Add autoplay
+  useInterval(
+    () => {
+      if (autoplay) {
+        goNext()
+      }
+    },
+    autoplay ? duration : null
+  )
+
+  useEffect(() => {
+    const handleKeys = (e: KeyboardEvent) => {
+      if (e.code === 'ArrowLeft') {
+        goPrev()
+      }
+
+      if (e.code === 'ArrowRight') {
+        goNext()
+      }
+    }
+
+    window.addEventListener('keydown', handleKeys)
```

Comment:
We should onkeydown only within the swiper container otherwise pressing the left/right arrow will make every swiper of the page navigating

**In production:**

https://github.com/user-attachments/assets/65293b81-6732-4aa8-a21d-e55a7352d75e

Reply:
Nice catch #yolo, on it :D

Reply:
I will create another issue to avoid breaking changes here

Reply:
https://linear.app/wttj/issue/WUI-225/swiper-keydown-on-all-swipers